spec/SPEC.md

```diff
@@ -89,259 +79,469 @@ produced the executable.
 its Artifact Identifier.
 
-Git currently supports two varieties of gitoids. One is based on SHA1 and is
-in common use. The other is based on SHA256 and has been very slow to garner
-adoption. The [gitoid URI spec][gitoid_uri] uses different prefixes,
+Git currently supports two varieties of GitOIDs. One is based on SHA-1 and is
+in common use. The other is based on SHA-256 and has been slow to garner
+adoption.
+
+Git's use of SHA-1 is additionally complicated by the fact that Git actually
+uses a variant of SHA-1 in newer versions, called SHA-1CD (where "CD" stands
+for "collision detection") which tries to detect attempts to engineer
+purposeful hash collisions in SHA-1 and subverts them by modifying the
+operation of the hash algorithm in those cases. Git calls this "SHA-1" and does
+not frequently distinguish use of the two similar but not equivalent hash
+algorithms.
+
+The [GitOID URI spec][gitoid_uri] uses different prefixes,
 `gitoid:blob:sha1` or `gitoid:blob:sha256`, to distinguish which algorithm is
-being used for computing the gitoid of a blob. This document adopts the gitoid
-URI prefixes to distinguish Artifact Identifier Types. This approach is
-anticipated to extend gracefully as git adopts new hash types in the future.
+being used for computing the GitOID of an artifact, subject to the knowledge
+that `gitoid:blob:sha1` may describe either SHA-1 or SHA-1CD depending on the
+version of Git being used.
 
-All subsequent references to mandatory identifier types in this document should
-be interpreted to mean the list:
+This document adopts the GitOID URI prefixes to distinguish Artifact Identifier
+Types. This approach is anticipated to extend gracefully as Git adopts new hash
+types in the future.
 
-- `gitoid:blob:sha1`
-- `gitoid:blob:sha256`
+Given the challenges with SHA-1 including:
 
-### Artifact Input Manifest
+- Its weakness as a hash algorithm today, with some attacks already being
+  known publicly which permit collisions in some contexts,
+- The fact that Git itself is expending effort to transition away from the use
+  of SHA-1 and toward SHA-256,
+- A concern that SHA-1 could in the next five years be subject to orders to
+  transition away by some world governments,
 
-An Artifact Input Manifest for an Artifact enumerates the inputs to the
-build tool that produced the artifact.  
+We have decided to only permit the use of SHA-256 as a hash algorithm for
+Artifact Identifiers for OmniBOR.
 
-Hereafter in the spec Artifact Input Manifest will simply be referred to as Input Manifest.
+We reserve the right to extend this list to support additional hash algorithms
+in the future, for example if SHA-256 is determined to be broken by future
+computing capabilities.
 
-A given Input Manifest utilizes precisely one Artifact Identifier Type.
+All subsequent references to Artifact Identifier types in this document should
+be interpreted to mean the list:
 
-#### Input Manifest Identifier
+- `gitoid:blob:sha256`
+
```

Comment:
I *really* like that even though we've collapsed down to a single Artifact Identifier Type we still preserve it as a list.